frontend/public/components/auto-complete.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+import { useDocumentListener, getLabelsAsString } from '@console/shared';
+import { fuzzyCaseInsensitive } from './factory/table-filters';
+import { TextInput } from '@patternfly/react-core';
+import { KeyEventModes } from '@console/shared/src/hooks/component-visible';
+
+const MAX_SUGGESTIONS = 5;
+
+const labelParser = (nodes) => {
+  const labels = _.map(nodes, getLabelsAsString);
+  return _.uniq(_.flatten(labels));
+};
+
+const suggestionBoxKeyHandler = {
+  Escape: KeyEventModes.HIDE,
+};
+
+const textInputKeyHandler = {
+  Escape: KeyEventModes.HIDE,
+  '/': KeyEventModes.FOCUS,
+};
+
+const AutocompleteInput: React.FC<AutocompleteInputProps> = (props) => {
+  const [suggestions, setSuggestions] = React.useState<string[]>();
+  const { visible, setVisible, ref } = useDocumentListener(suggestionBoxKeyHandler);
+  const { ref: textInput } = useDocumentListener(textInputKeyHandler);
+  const { textValue, setTextValue, onSuggestionSelect, placeholder, showSuggestions, data } = props;
+
+  const onSelect = (value: string) => {
+    onSuggestionSelect(value);
+    if (visible) {
+      setVisible(false);
+    }
+  };
+
+  const activate = () => {
+    if (!_.isEmpty(textValue.trim())) {
+      setVisible(true);
+    }
+  };
+
+  const handleInput = (input: string) => {
+    if (!_.isEmpty(input)) {
+      setVisible(true);
+    } else {
+      setVisible(false);
+    }
+    setTextValue(input);
+  };
+
+  React.useEffect(() => {
+    if (textValue !== '' && visible && showSuggestions) {
+      const processed = labelParser(data);
+      const filtered = processed
+        .filter((item) => fuzzyCaseInsensitive(textValue, item))
+        .slice(0, MAX_SUGGESTIONS - 1);
```

Comment:
Looks like you have an off-by-one error here if you mean to show `MAX_SUGGESTIONS`.

```suggestion
        .slice(0, MAX_SUGGESTIONS );
```